docker-images/kafka-based/kafka/scripts/kafka_run.sh

```diff
@@ -70,13 +70,17 @@ if [ "$STRIMZI_KRAFT_ENABLED" = "true" ]; then
   fi
 
 else
-  rm -f /var/opt/kafka/kafka-ready /var/opt/kafka/zk-connected 2> /dev/null
-  KEY_STORE=/tmp/kafka/cluster.keystore.p12
-  TRUST_STORE=/tmp/kafka/cluster.truststore.p12
-  KAFKA_OPTS="${KAFKA_OPTS} -javaagent:$(ls "$KAFKA_HOME"/libs/kafka-agent*.jar)=/var/opt/kafka/kafka-ready:/var/opt/kafka/zk-connected:$KEY_STORE:$CERTS_STORE_PASSWORD:$TRUST_STORE:$CERTS_STORE_PASSWORD"
-  export KAFKA_OPTS
+  KAFKA_READY=/var/opt/kafka/kafka-ready
+  ZK_CONNECTED=/var/opt/kafka/zk-connected
+  rm -f $KAFKA_READY $ZK_CONNECTED 2> /dev/null
```

Comment:
You define these variables only in this branch of the condition. But you use them everywhere. That seems wrong.

Reply:
They are passed just empty/not existing when KRaft mode is enabled, because the Kafka Agent doesn't need to handle those file in such a mode.

Reply:
They are not passed as empty - they are not defined, which seems quite hacky.

Reply:
Tbh I don't see it hacky. If you check the code in the `KafkaAgent`, in the end they are got as empty which is what we want. I am not sure it's really worth having lines like

```shell
KAFKA_READY=
ZK_CONNECTED=
```

in the KRaft condition if/else.

Reply:
If nothing else, it would make it clear that this is intentional.

Reply:
I'm quite surprised shell check does not complain about uninitialized variables.

Reply:
@scholzj I added them as empty in the KRaft mode if/else branch with comment explaining the why.